lib/Frontend/Frontend.cpp

```diff
@@ -552,6 +552,12 @@ bool CompilerInstance::setup(const CompilerInvocation &Invoke,
   if (LangOpts.EnableModuleLoadingRemarks) {
     Invocation.getSearchPathOptions().dump(LangOpts.Target.isOSDarwin());
   }
+  
+  // If the compiler instance is managed by an IDE, inform the source manager to avoid
+  // memory-mapping source files that are likely to be mutated in an editor.
+  if (LangOpts.DiagnosticsEditorMode) {
```

Comment:
Is there a more authoritative way of knowing that we're in an IDE? `DiagnosticsEditorMode` has quite specific semantics that we're stretching a bit with this condition.

> The Swift compiler has a setting (under LangOptions) called `DiagnosticsEditorMode`. When set, diagnostics should be customized for an interactive editor that can display and apply complex fix-its, and worry less about the appearance in build logs and command-line environments.

Maybe we should have a new flag for this, more akin to
https://github.com/swiftlang/swift/blob/cc8af1793be740f976206e46f8178c652ca48355/include/swift/Basic/LangOptions.h#L500
?

Reply:
Most of the comments around where `DiagnosticsEditorMode` is used refer to it as `EditorMode`. To me, it makes the most sense to have a single flag indicating we are in the context of an editor vs. having multiple ad-hoc flags for more specific tweaks.

I considered trying to rename `DiagnosticsEditorMode` in this PR, but was trying to keep the diff minimal.

That said, I'm happy to add a new flag or do the aforementioned rename in this PR if others think it's a good idea.

Reply:
Added a new `OpenSourcesAsVolatile` lang opt and updated the var names accordingly.